### PR TITLE
Modifs to cover multiple errors

### PR DESCRIPTION
**Pull request by iOiurson**
This PR fixes various little bugs:
- `split_range` does not crash when `start` does not includes an `!` (for instance `A1:Sheet1!A3`)
- `tR` only included rows up to 100. Extended to 1000.
- unknown named ranges don't raise exception
- added IRR `excellib` function
- changed a little the way `Cell`s and `Range`s choose to update or not, with the `need_update` field. This should avoid unnecessary `update_range`, but the `Cell` update case doesn't work perfectly yet (output results are not correct in some cases, i'm on it).

### CODE REVIEW

**Comment by iOiurson** [comment]
`Ranges` now don't hold any values when created from a `cellmap`. They only hold references to actual `Cells`.

**Comment by SergeBredin** [comment]
How much it changes execution time ?

**Comment by iOiurson** [comment]
From what i could see, on the file 224, nothing.
But they are cases that don't give the correct value of evaluation, because of the way Ranges updates are handled. I need to investigate on this, and the resulting eval time might change (in a good or bad way...)

**Comment by iOiurson** [comment]
Two new ideas are set up:
- **Spreadsheet.pending** (https://github.com/anthill/koala/pull/42/commits/80b9608e2a55bd6558756403fbf1e22373a335db)

This is a `Spreadsheet` level `dict` that is filled when `update_range` is called, to list for each `Range` being updated all cells that already have been passed through. The regular `need_update` check was not enough since sometimes "fake" circular references impeached the `need_update` of a `Cell` to be set to `False` before it was called a second time, leading to actual infinite loops.
Note that using this, a `Cell` can be evaluated multiple times before its `need_update` field is actually set to `False`.

- **RangeCore.apply()** (https://github.com/anthill/koala/pull/42/commits/5521b2a064cabb02d059d5a460599315ba4884af)

Before, two methods were used to determine wether to process operations on all `Cells` of a `Range` or simply on the "associated" `Cell`. This was represented by a choice between `apply_all` and `apply_one` during the `ASTNode.emit()`, depending on wether the parent `ASTNode` was a special function or not (see https://github.com/anthill/koala/pull/42/commits/5521b2a064cabb02d059d5a460599315ba4884af#diff-85c72660739076bf4061ecd380ecff9eL539).
This is not enough, since sometimes the `find_special_function()` might return `False`, leading to `apply_one()` when `apply_all()` is actually needed, in cases where you operate on `Ranges` that don't have any associated `Cells`.
To correct this, i introduced the `apply()` function, which decides between `apply_all` and `apply_one` checking on associated `Cells` at evaluation time. This still leaves the need of using `find_special_function()` at graph generation.

File: koala/ast/Range.py [hunk 1]
```diff
@@ -150,21 +158,27 @@ def sheet(self):
...
+                cell.value = new_values[index]
+        else:
+            for key, value in enumerate(self.order):
+                self[value] = new_values[key]
 
-    def reset(self):
-        for key in self.keys():
-            self[key] = None
+    @property
+    def Cells(self):
```
Comment (SergeBredin) [inline koala/ast/Range.py]: -> change the name

File: koala/ast/Range.py [hunk 2]
```diff
@@ -234,17 +248,49 @@ def get(self, row, col = None):
...
+                return None
+        elif range.type == "horizontal":
+            if (range.start[0], col) in range.order:
+                return range.cells[range.order.index((range.start[0], col))]
+            else:
+                return None
+        else:
+            raise ExcelError('#VALUE!', 'cannot use find_associated_cell on %s' % range.type)
+
+    @staticmethod
```
Comment (SergeBredin) [inline koala/ast/Range.py]: -> add a comment to describe function

File: koala/ast/Range.py [hunk 3]
```diff
@@ -234,17 +248,49 @@ def get(self, row, col = None):
                 return new_value
 
     @staticmethod
+    def find_associated_cell(ref, range):
```
Comment (SergeBredin) [inline koala/ast/Range.py]: -> add a comment to describe function

File: koala/ast/Range.py [hunk 4]
```diff
@@ -258,19 +304,43 @@ def find_associated_values(ref, first = None, second = None):
...
+                cell = None
+
+            if cell is not None:
+                return RangeCore.apply_one(func, self, other, ref)
+            else:
+                return RangeCore.apply_all(func, self, other, ref)
+        else:
+            return RangeCore.apply_all(func, self, other, ref)
+
+    @staticmethod
```
Comment (SergeBredin) [inline koala/ast/Range.py]: add a comment to describe function

File: koala/ast/graph.py [hunk 5]
```diff
@@ -275,117 +300,115 @@ def print_value_tree(self,addr,indent):
...
-            return self.Range('%s:%s' % (addr1, addr2))
-            # return self.evaluate_range(CellRange('%s:%s' % (addr1, addr2),sheet), False)
 
-    # @profile
     def update_range(self, range):
-        for key in range:
-            if range[key] is None:
-                addr = get_cell_address(range.sheet, key)
-                range[key] = self.evaluate(addr)
+        debug = False
```
Comment (SergeBredin) [inline koala/ast/graph.py]: Can you describe it in few lines of comments ?

**Pull request merged**